lib/handlers/unsubscribe.js

```diff
@@ -11,18 +11,24 @@ function UnsubscribeState (client, packet, finish, granted) {
 
 function handleUnsubscribe (client, packet, done) {
   var broker = client.broker
+  broker.persistence.removeSubscriptions(client, packet.unsubscriptions, function (err) {
+    if (err) {
+      return done(err)
+    }
 
-  if (packet.messageId) {
-    broker.persistence.removeSubscriptions(client, packet.unsubscriptions, function (err) {
-      if (err) {
-        return done(err)
-      }
-
-      actualUnsubscribe(client, packet, done)
-    })
-  } else {
     actualUnsubscribe(client, packet, done)
-  }
+  })
+  // if (packet.messageId) {
+  //   broker.persistence.removeSubscriptions(client, packet.unsubscriptions, function (err) {
+  //     if (err) {
+  //       return done(err)
+  //     }
+  //
+  //     actualUnsubscribe(client, packet, done)
+  //   })
+  // } else {
+  //   actualUnsubscribe(client, packet, done)
+  // }
```

Comment:
I think the check is there to handle some weird situation. It was probably a hack I put in :/.